go.mod

```diff
@@ -28,6 +28,8 @@ require (
 	github.com/spf13/viper v1.4.0
 	github.com/stretchr/testify v1.4.0
 	github.com/syndtr/goleveldb v1.0.1-0.20190318030020-c3a204f8e965
+	gitlab.com/NebulousLabs/fastrand v0.0.0-20181126182046-603482d69e40 // indirect
+	gitlab.com/NebulousLabs/go-upnp v0.0.0-20181011194642-3a71999ed0d3
```

Comment:
`gitlab.com/NebulousLabs/go-upnp` is a wrapper around https://github.com/huin/goupnp with some added code.